### ISSUE TITLE
Send raw html instead of url or path

### ISSUE THREAD

**Issue by maxdow** [fraserxu/electron-pdf#156]
Hi,

is it possible to send html data as a source instead of an url ? 

To do something like this : 

```
app.post('/pdfexport', function(req,res){
     //in memory mode
    exporter.createJob(req.body, target, options, jobOptions).then( job => {
      // 
    })  
})
```

**Comment by codecounselor** [comment]
I don't think this is possible but I have not spent any time investigating it.  The reason I say this is because the way it works now is that an Electron browser window is created and a URL is loaded into the browser.  If you find that the Electron API supports such an approach let me know and we can discuss options.
Otherwise, I suspect you will need to first write your html to a file.

**Comment by maxdow** [comment]
Thanks for your answer. 

It's not a huge deal for my project I think i can manage to write a temporary html file.

I don't know much about this part in electron but if you load an url, maybe you can load a base64 data string ?

**Comment by codecounselor** [comment]
Here is the docs: http://electron.atom.io/docs/api/browser-window/#winloadurlurl-options 
I haven't seen anything to support what you are suggesting, sorry.

**Comment by maxdow** [comment]
And do you think it's possible to inject html data throught win.webContents ?

**Comment by maxdow** [comment]
This one : https://github.com/electron/electron/issues/1146#issuecomment-76460592 about data encoding

**Comment by JoeChapman** [comment]
I'm also interested in this. I was going to create a small wrapper to save the html as a temp file but I'd prefer not to need to.
Slightly related: I would also like to be able to buffer the output to the response rather than save it to a file. My "wrapper" could perform that action too.

**Comment by maxdow** [comment]
The problem is that dataencoding may have some limitations in case of large files.

It seems you don't have to save the response to a file, buffering already exist
